packages/openhim/test/index.js

```diff
@@ -45,33 +69,326 @@ describe('execute', () => {
+      ],
+      active: true,
 
-  before(() => {
-    mockRequest = superagentMock(request, ClientFixtures);
-  });
+      deceasedBoolean: false,
 
-  it('posts to API and returns state', () => {
-    let state = {
-      configuration: {
-        username: 'hello',
-        password: 'there',
-        apiUrl: 'https://openhim.org/demo',
+      managingOrganization: {
+        reference: 'Organization/1',
       },
-    };
+    })(state);
+    expect(data).to.eql('User Authenticated Successfully');
+  });
+});
+
+describe('getTransactions', () => {
+  it('should GET all transctions', async () => {
+    testServer
+      .intercept({
+        path: '/transactions',
+        method: 'GET',
+      })
+      .reply(200, [{ _id: '001', clientID: '1234' }], { ...jsonHeaders });
 
-    return execute(event(fixtures.event.requestBody))(state).then(state => {
-      let lastReference = state.references[0];
+    const { data } = await getTransactions()(state);
+    expect(data[0].clientID).to.eql('1234');
+  });
 
-      // Check that the eventData made it's way to the request as a string.
-      expect(lastReference.params).to.eql(
-        JSON.stringify(fixtures.event.requestBody)
+  it('should GET a single transaction', async () => {
+    testServer
+      .intercept({
+        path: '/transactions/001',
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          _id: '001',
+          clientID: '1234',
+        },
+        { ...jsonHeaders }
       );
-    });
+
+    const { data } = await getTransactions({ transactionId: '001' })(state);
+    expect(data.clientID).to.eql('1234');
+    expect(data._id).to.eql('001');
+  });
+
+  it('should GET transactions with filters', async () => {
+    testServer
+      .intercept({
+        path: '/transactions',
+        method: 'GET',
+        query: {
+          filterLimit: 1,
+          filterPage: 0,
+          filterRepresentation: 'full',
+          filters: '{"response.status":"200"}',
+        },
+      })
+      .reply(200, [{ _id: '001', clientID: '1234' }], { ...jsonHeaders });
+
+    const { data } = await getTransactions({
+      filterLimit: 1,
+      filterPage: 0,
+      filterRepresentation: 'full',
+      filters: '{"response.status":"200"}',
+    })(state);
+    expect(data[0].clientID).to.eql('1234');
+  });
+});
+
+describe('getChannels', () => {
+  it('should GET all channels', async () => {
+    testServer
+      .intercept({
+        path: '/channels',
+        method: 'GET',
+      })
+      .reply(200, [{ _id: '0123', name: 'Channel Test' }], { ...jsonHeaders });
+
+    const { data } = await getChannels()(state);
+    expect(data[0].name).to.eql('Channel Test');
+  });
+
+  it('should GET a single channel', async () => {
+    testServer
+      .intercept({
+        path: '/channels/0123',
+        method: 'GET',
+      })
+      .reply(
+        200,
+        {
+          _id: '0123',
+          name: 'Channel Test',
+        },
+        { ...jsonHeaders }
```

Comment:
If you rebase against main, you should be able to remove all these JSON headers objects. Mock interceptors will now default json content type headers when appropriate